src/main/java/net/coderbot/iris/pipeline/HandRenderer.java

```diff
@@ -2,157 +2,181 @@
-
-//		return !(!gameRenderer.getRenderHand()
-//				|| camera.isThirdPerson()
-//					|| !(camera.getEntity() instanceof EntityPlayer)
-//						|| gameRenderer.getPanoramicMode()
-//							|| Minecraft.getMinecraft().gameSettings.hideGUI
-//								|| (camera.getEntity() instanceof EntityLiving && ((EntityLiving)camera.getEntity()).isPlayerSleeping())
-//                                    // TODO: SPECTATOR
-//									/*|| Minecraft.getMinecraft().gameMode.getPlayerMode() == GameType.SPECTATOR*/);
-	}
-
-	public boolean isHandTranslucent(InteractionHand hand) {
+    public static final HandRenderer INSTANCE = new HandRenderer();
+
+    private boolean ACTIVE;
+    private boolean renderingSolid;
+    public static final float DEPTH = 0.125F;
+
+    private void setupGlState(RenderGlobal gameRenderer, Camera camera, float tickDelta) {
+        Minecraft mc = Minecraft.getMinecraft();
+
+        GL11.glMatrixMode(GL11.GL_PROJECTION);
+        GL11.glLoadIdentity();
+        // We need to scale the matrix by 0.125 so the hand doesn't clip through blocks.
+        GL11.glScalef(1.0F, 1.0F, DEPTH);
+
+        // TODO: Anaglyph
+        /*if (this.mc.gameSettings.anaglyph) {
+            GL11.glTranslatef((float) (-(anaglyphChannel * 2 - 1)) * 0.07F, 0.0F, 0.0F);
+        }*/
+
+        if (mc.entityRenderer.cameraZoom != 1.0D) {
+            GL11.glTranslatef((float) mc.entityRenderer.cameraYaw, (float) (-mc.entityRenderer.cameraPitch), 0.0F);
+            GL11.glScaled(mc.entityRenderer.cameraZoom, mc.entityRenderer.cameraZoom, 1.0D);
+        }
+
+        Project.gluPerspective(mc.entityRenderer.getFOVModifier(tickDelta, false), (float) mc.displayWidth / (float) mc.displayHeight, 0.05F, mc.entityRenderer.farPlaneDistance * 2.0F);
+
+
+        if (mc.playerController.enableEverythingIsScrewedUpMode()) {
+            GL11.glScalef(1.0F, 2 / 3f, 1.0F);
+        }
+
+        GL11.glMatrixMode(GL11.GL_MODELVIEW);
+        GL11.glLoadIdentity();
+
+        // TODO: Anaglyph
+        /*if (mc.gameSettings.anaglyph) {
+            GL11.glTranslatef((float) (anaglyphChannel * 2 - 1) * 0.1F, 0.0F, 0.0F);
+        }*/
+
+        mc.entityRenderer.hurtCameraEffect(tickDelta);
+
+        if (mc.gameSettings.viewBobbing) {
+            mc.entityRenderer.setupViewBobbing(tickDelta);
+        }
+    }
+
+    private boolean canRender(Camera camera, RenderGlobal gameRenderer) {
+        Minecraft mc = Minecraft.getMinecraft();
+
+        return mc.entityRenderer.debugViewDirection <= 0 &&
+               mc.gameSettings.thirdPersonView == 0 &&
+               !camera.getEntity().isPlayerSleeping() &&
+               !mc.gameSettings.hideGUI &&
+               // TODO: SPECTATOR
+               // mc.playerController.currentGameType
+               !mc.playerController.enableEverythingIsScrewedUpMode();
+    }
+
+    public boolean isHandTranslucent(InteractionHand hand) {
         // TODO: Offhand
```

Comment:
@Caedis Anything obvious need to be done here for offhand support?

Reply:
https://github.com/GTNewHorizons/Angelica/blob/master/src/main/java/com/gtnewhorizons/angelica/dynamiclights/DynamicLights.java#L371-L386 to give you an example.